### PR TITLE
refactor: move env vars to OptimumRBLNConfig

### PR DESCRIPTION
**Pull request by rebel-eunji**
### 🚀 Summary of Changes

1. Move env vars to OptimumRBLNConfig
2. Change rbln_config in additional_config to optimum_overrides

## Notes
1. Refactor the fields - `optimum-rbln path, written by the platform hook and the config sync`

---

### 📌 Related Issues / Tickets

* Resolves #
* Related to #

---

### ✅ Type of Change

* [ ] 🚀 Release (`release`)
* [ ] ✨ Feature (`feature`)
* [ ] 🧠 Model support (`model`)
* [ ] 🧬 Core engine changes (`core`)
* [ ] 🛠 Bug fix (`fix`)
* [ ] ⚙️ Performance improvement (`perf`)
* [ ] 🔁 Refactor or code cleanup (`refactor`)
* [ ] 📄 Documentation (`docs`)
* [ ] ❓ Other (`other`): please describe

---

### 🧪 How to Test

1. Run `...`
5. Verify output: `...`
6. Edge case tested: `...`

---

### 📸 Screenshots / Logs (if applicable)

---

### 📋 Checklist

* [ ] PR title follows Conventional Commits format
* [ ] This PR is linked to an existing issue
* [ ] The test method is described, and the expected result is clearly stated
* [ ] Relevant documentation has been updated (if applicable)

---

### 💬 Notes

---

### CODE REVIEW

**Comment by rebel-eunji** [comment]
@rebel-cjlee 
I agree with that. I'll fix it.

**Comment by rebel-cjlee** [comment]
@rebel-eunji Thanks.

If you are ok with it, I would like to start the config interation now instead of waiting. This PR already tells me which fields the optimum config needs, and that was the part I was missing. String now looks faster than doing it in steps.

So: this PR settles the shape of `OptimumRBLNConfig`, and I take those fields as the starting point for folding them into `RBLNConfig`. Does that sound good?

File: vllm_rbln/optimum_config.py [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+
+    sampler: bool = True
+    """Use the customized RBLN sampler."""
+
+    optimum_overrides: dict[str, Any] = field(default_factory=dict)
+    """Entries for optimum-rbln's model config (its `rbln_config`), laid over
+    what vllm-rbln derives from the vLLM settings when the model is compiled.
+    With a pre-compiled model only the `device` entries apply."""
+
+    prefix_block_size: int | None = None
```
Comment (rebel-eunji) [inline vllm_rbln/optimum_config.py:75]: @rebel-jonghewk @rebel-cjlee 
I'll rename the prefix_block_size to sub_block_size to align with RBLNConfig in the following PR.

Comment (rebel-cjlee) [reply vllm_rbln/optimum_config.py:75]: As #1122 includes the integration along with introducing `OptimumRBLNConfig`, should we continue maintaining this PR?

Comment (rebel-eunji) [reply vllm_rbln/optimum_config.py:75]: Yes. I'll merge this PR ASAP.
